rpc/query_server.go

```diff
@@ -758,7 +759,6 @@ func (s *QueryServer) EthGetTransactionReceipt(hash eth.Data) (*eth.JsonTxReceip
 		}
 		return resp, nil
 	}
-	snapshot.Release()
```

Comment:
It is redundant. `snapshot.Release()` has been called here https://github.com/loomnetwork/loomchain/blob/3e71b475020089761b7f07f9a5fac15c39fbea6f/rpc/query_server.go#L746

Reply:
It's not redundant, it's explicitly released before we go rummaging through the block store so the snapshot isn't retained longer than necessary.

Reply:
In that case `defer snapshot.Release()` should be removed as it is pointless to call it as the snapshot will always be released before.

Reply:
No, the `defer snapshot.Release()` is not pointless, it ensures the snapshot is released in case an error or panic causes the current function to return before the non-deferred `snapshot.Release()` is called.

Reply:
True, fixed